src/apirequest.ts

```diff
@@ -173,8 +171,9 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
     parameters.context._options.rootUrl !== undefined &&
     options.url !== undefined
   ) {
-    const path = options.url.slice(parameters.context._options.rootUrl.length);
-    options.url = resolve.resolve(parameters.context._options.rootUrl, path);
+    const originalUrl = new URL(options.url);
+    const path = originalUrl.href.substr(originalUrl.origin.length);
+    options.url = new URL(path, parameters.context._options.rootUrl).href;
```

Comment:
small concern I have here, is that this approach would throw on an invalid URL, where as I believe the resolve would not:

```
> new URL('batman')
Uncaught TypeError [ERR_INVALID_URL]: Invalid URL: batman
    at onParseError (internal/url.js:256:9)
```

vs.,

```
> resolve('batman', '/batman')
'/batman'
```

Two options I see:

1. `console.warn`, and fallback to the original URL.
2. take this as a breaking change.

Reply:
From user perspective, it makes no difference when the invalid URL fail early as the new change, or inside the `.request<T>(..)` function. In my opinion I would consider option 2, take as a breaking change, because the error message is more specific than the original error inside `request`.

For example:
```javascript
var gmailApi = require('googleapis').google.gmail({ version: 'v1', rootUrl: 'batman' });

gmailApi.users.history.list({ userId: 'me' });
```

Old Error: `TypeError: Only absolute URLs are supported`
New Error: `TypeError [ERR_INVALID_URL]: Invalid URL: batman`

Let me know your thoughts @bcoe

Reply:
Hey @bcoe, let me know how should I move forward, thanks.